src/plugins/dataflow/components/dataflow-tile.tsx

```diff
@@ -26,6 +26,7 @@ interface IDataflowTileState {
   programRecordingMode: number; // TODO: convert to enum
   isPlaying: boolean;
   playBackIndex: number;
+  recordIndex: number; //# of ticks for record
```

Comment:
Does this need to be in state, i.e. should setting it trigger a re-render of the entire tile?